Mathlib/RingTheory/Polynomial/DegreeLT.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Kenny Lau
+-/
+
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.LinearAlgebra.Matrix.Block
+import Mathlib.Algebra.Polynomial.Div
+import Mathlib.Algebra.Polynomial.Taylor
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Polynomials with degree less than `n`
+
+This file contains the properties of the submodule of polynomials of degree less than `n` in a
+(semi)ring `R`, denoted `R[X]_n`.
+
+## Main definitions/lemmas
+
+* `degreeLT.basis R n`: a basis for `R[X]_n` the submodule of polynomials with degree `< n`,
+  given by the monomials `X^i` for `i < n`.
+
+* `degreeLT.basisProd R m n`: a basis for `R[X]_m × R[X]_n`, which is the sum of two instances of
+  the basis given above.
+
+* `degreeLT.addLinearEquiv R m n`: an isomorphism between `R[X]_(m+n)` and `R[X]_m × R[X]_n`,
+  given by the fact that the bases are both indexed by `Fin (m+n)`. This is used for the Sylvester
+  matrix, which is the matrix representing the Sylvester map between these two spaces, in a future
+  file.
+
+* `taylorLinear r n`: The linear automorphism induced by `taylor r` on `R[X]_n` which sends `X` to
+  `X + r` and preserves degrees.
+
+-/
+
+
+namespace Polynomial
+
+@[inherit_doc] scoped notation:9000 R "[X]_" n:arg => Polynomial.degreeLT R n
+
+namespace degreeLT
+
+variable {R : Type*} [Semiring R] {m n : ℕ} (i : Fin n) (P : R[X]_n)
+
+variable (R) in
+/-- Basis for `R[X]_n` given by `X^i` with `i < n`. -/
+noncomputable def basis (n : ℕ) : Basis (Fin n) R R[X]_n :=
+  .ofEquivFun (Polynomial.degreeLTEquiv R n)
+
+instance : Module.Finite R R[X]_n :=
+  .of_basis <| basis ..
+
+instance : Module.Free R R[X]_n :=
+  .of_basis <| basis ..
+
+lemma X_pow_mem (i : Fin n) : X ^ i.val ∈ R[X]_n :=
```

Comment:
```suggestion
lemma X_pow_mem_degreeLT {i : ℕ} (hi : i < n) : X ^ i ∈ R[X]_n :=
```
And doesn't this belong in an earlier file?

Reply:
it was originally used more, but since the creation of the object `xpow`, it has never been used again, so I've deleted this lemma and incorporated it into the definition of `xpow`. (and technically if you want to use it there is now the more convenient but perhaps less idiomatic `(xpow R n i).2`)